src/main/kotlin/no/nav/faktureringskomponenten/service/beregning/AntallBeregner.kt

```diff
@@ -10,55 +10,59 @@ private val log = KotlinLogging.logger { }
 
 class AntallBeregner {
     companion object {
-        fun antallForPeriode(fom: LocalDate, tom: LocalDate): BigDecimal {
-            val gjelderSammeMånedOgÅr = fom.year == tom.year && fom.monthValue == tom.monthValue
-            if (gjelderSammeMånedOgÅr) {
-                log.debug("Beregner antall for periode $fom og $tom på én og samme måned")
-                return beregnAntallForEnMåned(fom, tom)
-            }
-            log.debug("Beregner antall for periode $fom og $tom på flere måneder")
-            return beregnAntallForFlereMåneder(fom, tom)
-        }
-
-        private fun beregnAntallForFlereMåneder(fom: LocalDate, tom: LocalDate): BigDecimal {
-            var totalAngittAntall = BigDecimal.ZERO
-            var currentDate = fom
-
-            while (currentDate <= tom) {
-                val erFørsteMåned = fom == currentDate
-                val erSisteMåned = currentDate.year == tom.year && currentDate.month == tom.month
-                val totalAntallDagerForMåned = currentDate.lengthOfMonth()
-                val dagerIgjenAvMåneden = totalAntallDagerForMåned - (currentDate.dayOfMonth - 1)
+        fun antallForPeriode(datoFra: LocalDate, datoTil: LocalDate): BigDecimal {
+            val førsteMånedDager = datoFra.lengthOfMonth().toBigDecimal()
+            val sisteMånedDager = datoTil.lengthOfMonth().toBigDecimal()
 
-                val angittAntallForMåned =
-                    BigDecimal(dagerIgjenAvMåneden.toDouble() / totalAntallDagerForMåned.toDouble())
-                        .setScale(2, RoundingMode.HALF_UP)
+            val erSammeMånedOgÅr = datoFra.year == datoTil.year && datoFra.monthValue == datoTil.monthValue
+            val førsteMånedProsent = beregnProsentTilFørsteMåned(erSammeMånedOgÅr, førsteMånedDager, datoFra, datoTil)
+            val sisteMånedProsent = beregnProsentTilSisteMåned(erSammeMånedOgÅr, datoTil, sisteMånedDager)
+            val månederMellom = beregnProsentTilMånederMellomFørsteOgSiste(erSammeMånedOgÅr, datoFra, datoTil)
```

Comment:
Du får selv avgjøre om hva du vil gjøre, men etter min mening veldig unødvendig å ha private metoder på dette. I dette tilfellet er det etter min mening overbruk av private metoder. Denne koden er i mine øyne mye mer lesbar av å ha klare variabelnavn og assigne det i samme funksjon. Tror ikke du vil ha mer enn 2-3 ekstra linjer på denne funksjonen vedå fjerne alle private metodene. Da vil hele klassen være på mindre enn 30 linjer.

Meen jeg mistenker det kan bli en diskusjon ut av det, og jeg er ikke så keen på diskusjon her, føler dette kan dra seg langt, og jeg har ikke tid til det i dag. Så do what you wanna do.

Reply:
Jeg ser jeg har kommentert på en gammel commit - den nye versjonen ser bedre ut.